### DISCUSSION TITLE
Is the package `kmod` in rootfs necessary?

### DISCUSSION THREAD

**Discussion by yukiiiteru** [kata-containers/kata-containers#3357] [Q&A] [answered]
I've noticed that the `kmod` was added into rootfs of clearlinux and ubuntu in commit 495a92d2 for fixing #341, but in recent release static tar packages, the guest kernel config shows that the `CONFIG_MODULES` is disabled.

So I want to know that is it necessary to install `kmod` in rootfs?

**Comment by yukiiiteru** [comment]
In addition, the docs of NVIDIA GPU passthrough says that with this demand guest kernel should be recompiled.

My opinion is that if there is a need to passthrough GPU or use other features, the rootfs should also be regenerated with `kmod`, otherwise the `kmod` should not be installed by default.

**Comment by Jakob-Naucke** [comment]
cc @devimc

**Comment by devimc** [comment]
it's required for some custom guest kernels, see 

https://github.com/kata-containers/kata-containers/blob/cf464668ff196138462297023d8287c88ef8406c/tools/osbuilder/dockerfiles/QAT/qat.conf#L14

https://github.com/kata-containers/kata-containers/blob/cf464668ff196138462297023d8287c88ef8406c/tools/packaging/kernel/configs/fragments/gpu/nvidia.conf#L7

**Reply by yukiiiteru** [reply]
Thus, the `CONFIG_MODULES` is an optional config and the `kmod` should also be an optional package rather than a necessary package.

Is my understanding right?

**Reply by devimc** [reply, marked as answer]
right

**Reply by yukiiiteru** [reply]
Got it, thanks for your answer.
